src/command/defaults/ClearCommand.php

```diff
@@ -0,0 +1,174 @@
+	}
+
+	public function execute(CommandSender $sender, string $commandLabel, array $args){
+		if(!$this->testPermission($sender)){
+			return true;
+		}
+
+		if(count($args) > 3){
+			throw new InvalidCommandSyntaxException();
+		}
+
+		$target = null;
+		if(isset($args[0])){
+			if(!$sender->hasPermission("pocketmine.command.clear.other")){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender->getServer()->getPlayer($args[0]);
+			if($target === null){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.player.notFound"));
+				return true;
+			}
+		}elseif($sender instanceof Player){
+			if(!$sender->hasPermission("pocketmine.command.clear.self")){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender;
+		}else{
+			throw new InvalidCommandSyntaxException();
+		}
+
+		$item = null;
+		if(isset($args[1])){
+			try{
+				$item = LegacyStringToItemParser::getInstance()->parse($args[1]);
+			}catch(InvalidArgumentException $e){
+				//vanilla checks this at argument parsing layer, can't come up with a better alternative
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.give.item.notFound", [$args[1]]));
+				return true;
+			}
+		}
+
+		$maxCount = -1;
+		if(isset($args[2])){
+			$item->setCount($maxCount = $this->getInteger($sender, $args[2], 0));
+		}
+
+		//checking players inventory for all the items matching the criteria
+		if($item !== null and $maxCount === 0){
+			$count = 0;
+			$contents = array_merge($target->getInventory()->all($item), $target->getArmorInventory()->all($item));
+			foreach($contents as $content){
+				$count += $content->getCount();
+			}
+
+			if($count > 0){
+				Command::broadcastCommandMessage($sender, new TranslationContainer("%commands.clear.testing", [$target->getName(), $count]));
+			}else{
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.clear.failure.no.items", [$target->getName()]));
+			}
+
+			return true;
+		}
+
+		$cleared = 0;
+
+		//clear everything from the targets inventory
+		if($item === null){
+			$contents = array_merge($target->getInventory()->getContents(), $target->getArmorInventory()->getContents());
+			foreach($contents as $content){
+				$cleared += $content->getCount();
+			}
+
+			$target->getInventory()->clearAll();
+			$target->getArmorInventory()->clearAll();
+			$target->getCursorInventory()->clearAll();
+		}else{
+			//clear the item from targets inventory irrelevant of the count
+			if($maxCount === -1){
+				if(($index = $target->getArmorInventory()->first($item)) !== -1){
+					$cleared++;
+					$target->getArmorInventory()->clear($index);
+				}
+
+				foreach($target->getInventory()->all($item) as $index => $i){
```

Comment:
`$index` already exists outside the loop, so PHPStan doesn't like you overwriting it. You can use a different variable name to get around it.